sound/soc/sof/sof-priv.h

```diff
@@ -67,100 +67,110 @@ struct snd_sof_pdata;
  * and DSP device(s).
  */
 struct snd_sof_dsp_ops {
+
 	/* probe and remove */
-	int (*remove)(struct snd_sof_dev *sof_dev);
-	int (*probe)(struct snd_sof_dev *sof_dev);
+	int (*probe)(struct snd_sof_dev *sof_dev); /* mandatory */
+	int (*remove)(struct snd_sof_dev *sof_dev); /* optional */
 
 	/* DSP core boot / reset */
-	int (*run)(struct snd_sof_dev *sof_dev);
-	int (*stall)(struct snd_sof_dev *sof_dev);
-	int (*reset)(struct snd_sof_dev *sof_dev);
+	int (*run)(struct snd_sof_dev *sof_dev); /* mandatory */
+	int (*stall)(struct snd_sof_dev *sof_dev); /* optional */
+	int (*reset)(struct snd_sof_dev *sof_dev); /* mandatory */
```

Comment:
meh. not defined for HDaudio.

Reply:
@plbossart all 3 comments addressed and PR updated now.